src/signet.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2019-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_SIGNET_H
+#define BITCOIN_SIGNET_H
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#include <consensus/params.h>
+
+#include <stdint.h>
```

Comment:
In commit 7168395ea63d33075a60369f5778eb635208beba

Are all these includes really needed? I've removed all non-std-lib includes and everything compiles fine with this diff:

```diff
diff --git a/src/signet.h b/src/signet.h
index b9946f950a..679df85c5d 100644
--- a/src/signet.h
+++ b/src/signet.h
@@ -5,18 +5,16 @@
 #ifndef BITCOIN_SIGNET_H
 #define BITCOIN_SIGNET_H
 
-#if defined(HAVE_CONFIG_H)
-#include <config/bitcoin-config.h>
-#endif
-
-#include <consensus/params.h>
-
-#include <stdint.h>
+#include <cstdint>
+#include <vector>
 
 class CBlock;
 class CScript;
 class uint256;
 struct CMutableTransaction;
+namespace Consensus {
+struct Params;
+}
 
 constexpr uint8_t SIGNET_HEADER[4] = {0xec, 0xc7, 0xda, 0xa2};

Reply:
Not sure about the namespace/struct part, but otherwise sounds reasonable. Updated to this.